### PR TITLE
fix issue 199 - add containerized and non-containerized ability to Makefile

### LINKED ISSUE #199: Containerized only build breaks downstream
Containerizing the build process removed the ability to do a non-containerized build. Can the ability to do a non-containerized build be put back into the makefile so that the option is available?

### PR DESCRIPTION
**Pull request by screeley44**
fixes #199

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -53,7 +58,15 @@ LDFLAGS='-extldflags "-static"'
 controller-bin:
 	@echo '********'
 	@echo 'Compiling controller binary'
-	docker run -it --rm -v $(REPO_ROOT):$(WORK_DIR):Z -w $(WORK_DIR) -e GOOS=$(GOOS) -e GOARCH=$(ARCH) -e CGO_ENABLED=$(CGO_ENABLED) $(BUILD_IMAGE) go build -a -ldflags $(LDFLAGS) -o $(WORK_DIR)/bin/$(CONTROLLER_BIN) $(WORK_DIR)/cmd/controller/controller.go
+	$(eval DOCKER_CMD_COUNT=$(shell docker ps | wc -c))
```
Comment (jeffvance) [inline Makefile]: I think it is sufficient to just check if `docker ps` works. There do not have to be running containers in order for us to decide to use a containerized make.

**Comment by jeffvance** [comment]
Where is the non-containerized build for `importer-bin` and the containerized build for `test-bin`?

File: Makefile [hunk 2]
```diff
@@ -33,6 +34,10 @@ IMPT_IMG_NAME=cdi-$(IMPORTER)
 GIT_USER=$(shell git config --get user.email | sed 's/@.*//')
 TAG=$(GIT_USER)-latest
 
+# Preflight Checks
+USE_DOCKER=1
+DOCKER_CMD_COUNT=0
```
[the remark this answers is not in the document]
Comment (screeley44) [reply Makefile]: how do you plan on evaluating `docker ps` ? if docker isn't running the wc returns 0 if it is it's > 0, and I need a way to pass that back into my evaluation. If I use exit code, I'm still going to have to create a variable to store that information and access again won't I?

File: Makefile [hunk 3]
```diff
@@ -53,7 +58,15 @@ LDFLAGS='-extldflags "-static"'
 controller-bin:
 	@echo '********'
 	@echo 'Compiling controller binary'
-	docker run -it --rm -v $(REPO_ROOT):$(WORK_DIR):Z -w $(WORK_DIR) -e GOOS=$(GOOS) -e GOARCH=$(ARCH) -e CGO_ENABLED=$(CGO_ENABLED) $(BUILD_IMAGE) go build -a -ldflags $(LDFLAGS) -o $(WORK_DIR)/bin/$(CONTROLLER_BIN) $(WORK_DIR)/cmd/controller/controller.go
+	$(eval DOCKER_CMD_COUNT=$(shell docker ps | wc -c))
+	export DOCKER_CMD_COUNT
```
Comment (copejon) [inline Makefile]: L61 should have set this var in the env already

File: Makefile [hunk 4]
```diff
@@ -53,7 +58,15 @@ LDFLAGS='-extldflags "-static"'
...
+	@if [ $(DOCKER_CMD_COUNT) -eq 0 ] || [ $(USE_DOCKER) -eq 0 ]; then \
```
Comment (copejon) [inline Makefile]: it _may_ be cleaner to use make if/else blocks rather than in shell.

File: Makefile (hunk 3 above)
Comment (screeley44) [reply Makefile]: not that I can access from within that little context of the makefile, the only way I could get DOCKER_CMD_COUNT was to export, it's strange, but basically if I don't export and then try to access that var, I get empty

File: Makefile (hunk 1 above)
Comment (screeley44) [reply Makefile]: this does not require containers to be present, if docker is not running it evaluates correctly, if docker is not installed it evaluates correctly, and it does not need containers to be installed.

File: Makefile (hunk 4 above)
Comment (screeley44) [reply Makefile]: I've tried, and the `ifeq else endif` blocks didn't work, they errored and/or they never evaluated the vars correctly. that is why I moved to the shell, but probably the bigger issue I didn't like is there is no complex conditional support (if AND/OR) so we would have to nest our ifs

File: Makefile [hunk 5]
```diff
@@ -33,6 +35,12 @@ IMPT_IMG_NAME=cdi-$(IMPORTER)
 GIT_USER=$(shell git config --get user.email | sed 's/@.*//')
 TAG=$(GIT_USER)-latest
 
+# Preflight Check Defaults
+USE_DOCKER=1
+DOCKER_CMD_COUNT=0
+$(eval DOCKER_CMD_COUNT=$(shell docker ps | wc -c))
+export DOCKER_CMD_COUNT
```
Comment (copejon) [inline Makefile]: mixing make and shell syntax

File: Makefile [hunk 6]
```diff
@@ -33,6 +39,11 @@ IMPT_IMG_NAME=cdi-$(IMPORTER)
 GIT_USER=$(shell git config --get user.email | sed 's/@.*//')
 TAG=$(GIT_USER)-latest
 
+# Preflight Check Defaults
+USE_DOCKER=1
+$(eval DOCKER_OUT=$(shell docker ps))
```
Comment (copejon) [inline Makefile]: This mixes make syntax (everything left justified) and bash syntax (everything under a target indented by 1).

To set the variable globally in the Makefile, you can do
DOCKER_OUT=$(shell docker ps)
DOCKER_OUT != docker ps

File: Makefile [hunk 7]
```diff
@@ -33,6 +39,11 @@ IMPT_IMG_NAME=cdi-$(IMPORTER)
...
+export DOCKER_OUT
```
Comment (copejon) [inline Makefile]: don't need to export if set as defined above

Comment (copejon) [reply Makefile]: also, not sure this would work, since it's bash syntax

Comment (screeley44) [reply Makefile]: doesn't like the DOCKER_OUT != docker ps

**Pull request merged**